fundamentals/fundamentals-bio/src/main/scala/izumi/functional/bio/BIOParallel.scala

```diff
@@ -0,0 +1,41 @@
+package izumi.functional.bio
+
+trait BIOParallel[F[-_, +_, +_]] extends BIORoot {
+  val InnerF: BIOFunctor3[F]
+
+  def parTraverseN[R, E, A, B](maxConcurrent: Int)(l: Iterable[A])(f: A => F[R, E, B]): F[R, E, List[B]]
+  def parTraverse[R, E, A, B](l: Iterable[A])(f: A => F[R, E, B]): F[R, E, List[B]]
+
+  /**
+    * Returns an effect that executes both effects,
+    * in parallel, combining their results with the specified `f` function. If
+    * either side fails, then the other side will be interrupted.
+    */
+  def zipWithPar[R, E, A, R1 <: R, E1 >: E, B, C](fa: F[R, E, A], fb: F[R1, E1, B])(f: (A, B) => C): F[R1, E1, C]
+
+  // defaults
+  def parTraverse_[R, E, A, B](l: Iterable[A])(f: A => F[R, E, B]): F[R, E, Unit] = InnerF.void(parTraverse(l)(f))
```

Comment:
I have noticed that `parTraverse_` is not `final`. Should this be made `final`? If not, why?

Reply:
@Adriani277
This shouldn't be made `final` because `BIOAsyncZio` instance overrides it to use `ZIO.foreachPar_`. This lets it use a native implementation from the effect type that is more optimized and may have enhancements compared to a naive implementation (like, better support for stack-traces, e.g. notice the `new ZipLeftFn` wrapper in implementation of `ZIO.<*` - https://github.com/zio/zio/blob/master/core/shared/src/main/scala/zio/ZIO.scala#L108 - this allows the stacktrace where `<*` occurs to point to the source code line contained inside the user-provided `that` lambda, instead of the line in ZIO.scala source file that would be there without the `new ZipLeftFn`)

Generally, the only methods that are final in the hierarchy are those that are   trivial enough they _can't_ have better native implementations in the effect type. So, I'd propose to un-final `zipParLeft` & `zipParRight` and redirect them to `<&` and `&>` in `BIOAsyncZio`

Reply:
I see, thanks for the explanation.

Will make the changes